JOSS Review (@rgmyr): Release build fails

`make` failed to build after configuring with `-DCMAKE_BUILD_TYPE=Release`. I also tried adding a `-Wno-error` flag to the `make` command, but that didn't seem to have any effect. 

I've attached the output: [Release_build_make_output.txt](https://github.com/kellykochanski/rescal-snow/files/3601982/Release_build_make_output.txt). It fails after building/linking `zlib` objects, so maybe this is related to my other issue.

I was able to build after running `cmake` with `-DCMAKE_BUILD_TYPE=Debug`, after which the example scripts seem to be working as expected.

@egreen77 I now get the following build error:

```bash
make[2]: *** No rule to make target '../lib/zlib-1.2.11/libz.a', needed by 'bin2png'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 94%] Building C object CMakeFiles/bin2png.dir/src/bin2png.c.o
make[2]: *** No rule to make target '../lib/zlib-1.2.11/libz.a', needed by 'csp2png'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 94%] Building C object CMakeFiles/csp2png.dir/src/csp2png.c.o
[ 95%] Building C object CMakeFiles/rescal.dir/src/drand48.c.o
[ 96%] Building C object CMakeFiles/rescal.dir/src/entry.c.o
make[2]: *** No rule to make target '../lib/zlib-1.2.11/libz.a', needed by 'rescal'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 96%] Building C object CMakeFiles/rescal.dir/src/rescal.c.o
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/bin2png.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:295: CMakeFiles/rescal.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:147: CMakeFiles/csp2png.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/src/format.c
+++ b/src/format.c
@@ -257,7 +257,11 @@ void read_csp(char *filename) {
   LogPrintf("csp_cell_size = %d\n", csp_cell_size);
 
   for (k = csp_y_bounds; k < D - csp_y_bounds; k++) { // profondeur
-    fread(buf, csp_cell_size, csp_H * csp_L, fp);
+    if ((int32_t)fread(buf, csp_cell_size, csp_H * csp_L, fp) != (csp_cell_size * csp_H * csp_L)) {
+      ErrPrintf("Read error\n");
+      exit(-1);
+    }
+
     aux = buf;
     pt = TE + k * HL;
     for (j = 0; j < H; j++) { // hauteur

--- a/src/genesis.c
+++ b/src/genesis.c
@@ -666,7 +666,10 @@ float* load_surface(char *nom, int32_t lx, int32_t ly) {
   pt_al = altimap;
   for (j = 0; j < ly; j++)
     for (i = 0; i < lx; i++, pt_al++) {
-      fscanf(f, "%f \n", pt_al);
+      if (fscanf(f, "%f \n", pt_al) != 1) {
+        ErrPrintf("Read error\n");
+        exit(-1);
+      }
     }
 
   fclose(f);

--- a/src/lgca.c
+++ b/src/lgca.c
@@ -566,7 +566,10 @@ void lecture_mvt() {
     exit(-4);
   }
 
-  fread(CelMvt, sizeof(MvtField), CHLD, fp);
+  if (fread(CelMvt, sizeof(MvtField), CHLD, fp) != sizeof(MvtField) * CHLD) {
+    ErrPrintf("Read error\n");
+    exit(-1);
+  }
 
   fclose(fp);
 

